src/rules/noVar.ts

```diff
@@ -0,0 +1,42 @@
+import { generate } from 'astring'
+import * as es from 'estree'
+
+import { ErrorSeverity, ErrorType, Rule, SourceError } from '../types'
+
+export class NoVarError implements SourceError {
+  public type = ErrorType.SYNTAX
+  public severity = ErrorSeverity.ERROR
+
+  constructor(public node: es.VariableDeclaration) {}
+
+  get location() {
+    return this.node.loc!
+  }
+
+  public explain() {
+    return 'Variable declaration using "var" is not allowed.'
+  }
+
+  public elaborate() {
+    const name = (this.node.declarations[0].id as es.Identifier).name
+    const value = generate(this.node.declarations[0].init)
+
+    return `Use keyword "let" instead, to declare a variable:\n\n\tlet ${name} = ${value};`
+  }
+}
+
+const noVar: Rule<es.VariableDeclaration> = {
+  name: 'no-var',
+
+  checkers: {
+    VariableDeclaration(node: es.VariableDeclaration, ancestors: [es.Node]) {
+      if (node.kind === 'var') {
```

Comment:
I think it might be better to have a rule to allow const and let only rather than to disallow var, and we can merge with noDeclareMutable

Reply:
Nevermind, I take this back. Not currently sane to have a whitelist using rules.